File: CSharp/PaymentsService/Infrastructure/Statistics/StatisticsService.cs
using PaymentsService.Domain.Payments;
using PaymentsService.Domain.Statistics;

namespace PaymentsService.Infrastructure.Statistics;

public class StatisticsService : IStatisticsService
{
    private readonly List<object> _statistics = new();

    public void CountStatistics(ProcessPaymentBaseRequest request, ProcessPaymentBaseResponse response)
    {
        _statistics.Add(new {request, response});
    }
}

File: Python/PaymentsService/Infrastructure/Statistics/StatisticsService.py
from Domain.Statistics.IStatisticsService import IStatisticsService
from typing import List

class StatisticsService(IStatisticsService):
    def __init__(self):
        self._statistics = []

    def count_statistics(self, request, response):
        self._statistics.append({'request': request, 'response': response})

